### ISSUE TITLE
(x64) Glide64 Build Errors

### ISSUE THREAD

**Issue by tony971** [project64/project64#680]
https://gist.github.com/tony971/655a5298f7dca1341608

**Comment by tony971** [comment]
Updated errors after a clean clone.

**Comment by cxd4** [comment]
That's the linker error I've started to get too, but not with 64-bit because I do not build PJGlide64 when trying to update the Project64 core.  So I leave the option to build Glide unchecked.

Were you intending on supplying the SDL link-time library with the repository @project64, or could it be preferable for us to attempt elimination of the SDL dependency for Win32 builds?

**Comment by project64** [comment]
@cxd4 maybe the code can be compiled for windows with out SDL I was just trying to sync the code, if I go back over it maybe I can remove the dependancy

**Comment by cxd4** [comment]
Alright, then in the meantime I will link to it for you:
http://libsdl.org/release/SDL2-devel-2.0.3-VC.zip

`SDL2-2.0.3\lib\x86\SDL2.lib` can go to `$project64\Bin\Debug\lib`.
With this the Debug Win32 build succeeds without errors, and the rest of us can use that to test future changes involved with possibly removing the SDL dependency.

**Comment by tony971** [comment]
Still happens after #681

**Comment by cxd4** [comment]
Are you using the x64 SDL2.lib from that .zip I linked?

Because I have never seen that compiler error you pasted before, looks new to me.

Then again the error my 681 pull request fixed didn't happen for you or zilmar either, just me.  I think these SDL 2.0 wip changes are disregarding backwards compatibility.

**Comment by tony971** [comment]
I wasn't but the error also occurs with your lib file. Maybe it's a VS15 thing?

**Comment by project64** [comment]
I can compile on vs2013 fine so probably a VS2015. I know @cxd4 removed some (all) the dependency on sdl. Which if it does have no dependency at the moment you could try to remove the reference to sdl

**Comment by cxd4** [comment]
I'll look for remaining SDL references to remove.

My original intent was to remove them all but then I started to think that keeping in sync with upstream Glide64 was a greater priority so meant to just remove the one that caused my build to fail.  But I'll look for others at this point.

**Comment by tony971** [comment]
Upstream Glide64? People are still working on that? I thought Gonetz was the last one.

**Comment by cxd4** [comment]
There are a lot of changes/improvements merged from the cross-platform / Android fork so Glide64 may perform better/worse now for many games.  I think @AmbientMalice was one of the ones who reported issues/regressions with PJGlide64 fork of Glide64 so now with the major modifications some of those past issues might use some reviewing.  Though I think mostly it will be the same.

**Comment by JunielKatarn** [comment]
This PR fixes the issue: #687

The issue seems to be caused by SDL source file src/stdlib/SDL_stdlib.c (247):
```
#if _MSC_VER >= 1600 && defined(_WIN64) && !defined(_DEBUG)
#include <intrin.h>

#pragma function(memcpy)
void * memcpy ( void * destination, const void * source, size_t num )
```

In short, that file redefines memcpy for the x64|Release profie, for some reason I don't understand. (if I suppress that condition, PJ64 build fine).

On another note, Is there a 64-bit audio plugin I can use to test the x64 build?

**Comment by LegendOfDragoon** [comment]
>On another note, Is there a 64-bit audio plugin I can use to test the x64 build?

Azimer's should work, iirc.

**Comment by cxd4** [comment]
SDL does a hell of a lot of re-inventing the wheel.
Like all of those standard C functions like memcpy--SDL practically creates its own platform.

It even does unnecessary garbage like assembly language, for 16-bit byte-swapping, which is perfectly well optimized to a ROL or XCHG by C compilers as old as VC2005.

>On another note, Is there a 64-bit audio plugin I can use to test the x64 build?

You can compile [Azimer's Audio](https://github.com/Azimer/AziAudio) or use the No Sound plugin I released.

**Comment by tony971** [comment]
Fixed after #687
